drivers/mrf24j40/Makefile.dep

```diff
@@ -1,11 +1,14 @@
+USEMODULE += ztimer_usec
 USEMODULE += ieee802154
-USEMODULE += netdev_ieee802154
-USEMODULE += netdev_legacy_api
-USEMODULE += xtimer
+USEMODULE += bhp
+
 FEATURES_REQUIRED += periph_gpio
 FEATURES_REQUIRED += periph_gpio_irq
 FEATURES_REQUIRED += periph_spi
 
-ifneq (,$(filter mrf24j40m%,$(USEMODULE)))
-  DEFAULT_MODULE += netdev_ieee802154_oqpsk
+ifneq (,$(filter netdev,$(USEMODULE)))
+  USEMODULE += netdev_ieee802154_submac
+  ifneq (,$(filter mrf24j40m%,$(USEMODULE)))
+    DEFAULT_MODULE += netdev_ieee802154_oqpsk
```

Comment:
This I don't understand. The module type should only influence the PA/LNA settings, `netdev_ieee802154_oqpsk` is a property of the chip. (I know this was there before, but still…)

Reply:
AFAIU this is just for signaling that the transceiver supports `O-QPSK`. This will probably disappear as soon as all radios implement the HAL.

Reply:
But then why make it conditional to `mrf24j40m%` which is used for `mrf24j40ma` - `mrf24j40mc` (modules with and without external PA/LNA)

Reply:
It was probably a mistake. I can just include it unconditionally

Reply:
done